runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -692,23 +692,47 @@ TR_VectorAPIExpansion::getDataTypeFromClassNode(TR::Compilation *comp, TR::Node
 
    if (!j9class) return TR::NoType;
 
-   TR_J9VMBase *fej9 = comp->fej9();
-   J9JavaVM *vm = fej9->getJ9JITConfig()->javaVM;
-
-   if (j9class == vm->floatReflectClass)
-      return TR::Float;
-   else if (j9class == vm->doubleReflectClass)
-      return TR::Double;
-   else if (j9class == vm->byteReflectClass)
-      return TR::Int8;
-   else if (j9class == vm->shortReflectClass)
-      return TR::Int16;
-   else if (j9class == vm->intReflectClass)
-      return TR::Int32;
-   else if (j9class == vm->longReflectClass)
-      return TR::Int64;
+#if defined(J9VM_OPT_JITSERVER)
```

Comment:
This is correct, though I think it would be better if we can avoid sprinkling this `#if defined(J9VM_OPT_JITSERVER)` throughout the code base. What I am proposing is to create another frontend query, say
```
TR::DataType
TR_J9VMBase::getClassPrimitiveDataType(TR_OpaqueClassBlock* clazz)
   {
   J9Class *j9class = TR::Compiler->cls.convertClassOffsetToClassPtr(clazz);
   if (!j9class) 
      return TR::NoType;

   J9JavaVM *vm = getJ9JITConfig()->javaVM;

   if (j9class == vm->floatReflectClass)
      return TR::Float;
   else if (j9class == vm->doubleReflectClass)
      return TR::Double;
   else if (j9class == vm->byteReflectClass)
      return TR::Int8;
   else if (j9class == vm->shortReflectClass)
      return TR::Int16;
   else if (j9class == vm->intReflectClass)
      return TR::Int32;
   else if (j9class == vm->longReflectClass)
      return TR::Int64;
   else
      return TR::NoType;
   }
```
and override it at JITServer in `TR_J9ServerVM`. Here you can use:
`auto *vmInfo = _compInfoPT->getClientData()->getOrCacheVMInfo(stream)`

Reply:
To confirm, what we will basically do is to declare two new methods: `TR_J9VMBase::getClassPrimitiveDataType` and `TR_J9ServerVM::getClassPrimitiveDataType` ,  and change `TR_J9VMBase::getPrimitiveArrayTypeCode` so that it is just `return getClassPrimitiveDataType(...)`

Reply:
`TR_J9VMBase::getPrimitiveArrayTypeCode` has a different return type. I am not sure if you can combine the two